src/FxKit/Option/Option.NaturalTransformations.cs

```diff
@@ -24,9 +25,30 @@ public static partial class Option
         Func<R> selector)
         where T : notnull
         where R : notnull =>
-        source.Match(
-            Some: Result<T, R>.Ok,
-            None: () => Err<T, R>(selector()));
+        source.TryGet(out var value)
+            ? Result<T, R>.Ok(value)
+            : Result<T, R>.Err(selector());
+
+    /// <summary>
+    /// Returns a result containing the value from an <see cref="Option{T}"/> if it exists.
+    /// Otherwise, returns a result with the specified error value.
+    /// </summary>
+    /// <typeparam name="T">The type of the value in the result.</typeparam>
+    /// <typeparam name="R">The type of the error in the result.</typeparam>
+    /// <param name="source">The input <see cref="Option{T}"/>.</param>
+    /// <param name="invalidValue">The default error value to return if the option is empty.</param>
+    /// <returns>A <see cref="Result{TOk,TErr}"/> containing either the value or the specified error.</returns>
+    [DebuggerHidden]
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    [GenerateTransformer]
+    public static Result<T, R> OkOrElse<T, R>(
```

Comment:
This should be named `OkOr` to be consistent with the other names that have both a materialized and a selector fallback variant.

Reply:
Nice, fixed